### ISSUE TITLE
Add support for subdirectories

### ISSUE THREAD

**Issue by da-h** [yaa110/nomino#2]
First, thanks for the work you've put into this project. I find it really handy. ;)

**Is your feature request related to a problem? Please describe.**
I think your project could benefit from letting the user also recurse into subdirectories

For, instance, one could replace a whole subtree-structure as follows:
```
➜ nomino -p -r "(.*)/(.*)/(.*)" "{} {}.{}"
```

Could restore your default structure
```
Nomino (2020) S1.E1.1080p.mkv
Nomino (2020) S1.E2.1080p.mkv
Nomino (2020) S1.E3.1080p.mkv
Nomino (2020) S1.E4.1080p.mkv
Nomino (2020) S1.E5.1080p.mkv
```

from, let's say a structure like this:
``` 
Nomino (2020)/S1/E1.1080p.mkv
Nomino (2020)/S1/E2.1080p.mkv
Nomino (2020)/S1/E3.1080p.mkv
Nomino (2020)/S1/E4.1080p.mkv
Nomino (2020)/S1/E5.1080p.mkv
```

**Describe the solution you'd like**
I would recommend that recursion is disabled by default, however, if a `/` is used, it is turned on automatically. Recursion depth should be an option in the cli as well.

**Additional context**
-

Best,
*da-h*

**Comment by yaa110** [comment]
Thank you for your feedback, just to clarify, do you need to support subdirectories in input or output?

**Comment by da-h** [comment]
I think it is best, if a user does not have to think about internals. That said
i think, in a more general setting, both are nice to have.

These feature would fit very well into the api you already provide

**Comment by yaa110** [comment]
Implemented in [v0.3.1](https://github.com/yaa110/nomino/releases/tag/0.3.1):

```bash
➜  tree
.
├── s1
│   ├── Nomino (2020) S5.E1.1080p.mkv
│   ├── Nomino (2020) S5.E2.1080p.mkv
│   ├── Nomino (2020) S5.E3.1080p.mkv
│   └── Nomino (2020) S5.E4.1080p.mkv
└── s2
    ├── Nomino (2020) S5.E1.1080p.mkv
    ├── Nomino (2020) S5.E2.1080p.mkv
    └── Nomino (2020) S5.E3.1080p.mkv

➜  nomino -r "s(\d+)/.*E(\d+).*" -kp "Season {:2}/Episode {:2}.mkv"
+----------------------------------+--------------------------+
| Input                            | Output                   |
+----------------------------------+--------------------------+
| s1/Nomino (2020) S5.E1.1080p.mkv | Season 01/Episode 01.mkv |
| s1/Nomino (2020) S5.E2.1080p.mkv | Season 01/Episode 02.mkv |
| s1/Nomino (2020) S5.E3.1080p.mkv | Season 01/Episode 03.mkv |
| s2/Nomino (2020) S5.E1.1080p.mkv | Season 02/Episode 01.mkv |
| s2/Nomino (2020) S5.E2.1080p.mkv | Season 02/Episode 02.mkv |
| s2/Nomino (2020) S5.E3.1080p.mkv | Season 02/Episode 03.mkv |
| s2/Nomino (2020) S5.E4.1080p.mkv | Season 02/Episode 04.mkv |
+----------------------------------+--------------------------+
```

**Comment by da-h** [comment]
Wow. Thanks a lot for implementing this feature into your tool.
I am looking forward to using it an a regular basis. ;)

Best
*da-h*
